juriscraper/scotus/scotus_docket_report.py

```diff
@@ -0,0 +1,413 @@
+        case_name = (
+            f"{petitioner_title} v. {respondent_title}"
+            if respondent_title
+            else petitioner_title
+        )
+        lower_court = scotus_data.get("LowerCourt", "")
+        lower_court_case_numbers_raw = scotus_data.get(
+            "LowerCourtCaseNumbers", ""
+        )
+        lower_court_decision_date = scotus_data.get("LowerCourtDecision", "")
+        lower_court_rehearing_denied_date = scotus_data.get(
+            "LowerCourtRehearingDenied", ""
+        )
+        links = scotus_data.get("Links", "").replace("Linked with", "").strip()
+        return {
+            "docket_number": docket_number,
+            "capital_case": scotus_data.get("bCapitalCase"),
+            "date_filed": self.normalize_date(docket_date),
+            "case_name": case_name.strip() if case_name else "",
+            "links": links,
+            "lower_court": lower_court.strip() if lower_court else None,
+            "lower_court_case_numbers_raw": lower_court_case_numbers_raw.strip()
+            if lower_court_case_numbers_raw
+            else None,
+            "lower_court_case_numbers": self.clean_lower_court_cases(
+                lower_court_case_numbers_raw
+            )
+            if lower_court_case_numbers_raw
+            else None,
+            "lower_court_decision_date": self.normalize_date(
+                lower_court_decision_date
+            ),
+            "lower_court_rehearing_denied_date": self.normalize_date(
+                lower_court_rehearing_denied_date
+            ),
+            "discretionary_court_decision": self.normalize_date(
+                scotus_data.get("DiscretionaryCourtDecision", "")
+            ),
+        }
+
+    @property
+    def docket_entries(self) -> list[dict[str, Any]]:
+        """Return a dictionary of docket entries from ProceedingsandOrder
+
+        [
+           {
+              "attachments":[
+                 {
+                    "document_url":"http://www.supremecourt.gov/DocketPDF/24/24A100/319653/20240723152722701_Johnson%20cert%20extension.pdf",
+                    "short_description":"Main Document"
+                 }
+              ],
+              "date_filed":"2024-07-30",
+              "description":"Application (24A100) granted by Justice Kagan extending the time to file until October 5, 2024.",
+              "description_html":"Application (24A100) <a href=#>granted</a> by Justice Kagan extending the time to file until October 5, 2024."
+           }
+        ]
+
+        :return: List of docket entry dicts
+        """
+        entries = []
+        for row in self._scotus_json.get("ProceedingsandOrder", []):
+            links = row.get("Links") or []
+            attachments = [
+                {
+                    "short_description": link.get("Description", "").strip(),
+                    "document_url": link.get("DocumentUrl"),
+                }
+                for link in links
+            ] or []
```

Comment:
I don't think the `or []` is necessary here. When the `links` array is empty, the list comprehension will produce an empty list of attachments.